ext/charlock_holmes/extconf.rb

```diff
@@ -50,7 +52,13 @@ def sys(cmd)
 have_library 'icuuc' or abort 'libicuuc missing'
 have_library 'icudata' or abort 'libicudata missing'
 
-$CXXFLAGS << ' -std=c++11'
+# icu4c might be built in C++11 mode, but it also might not have been
+icuconfig = `which icu-config`.chomp
+icuconfig = "#{icu4c}/bin/icu-config" if icuconfig.empty?
+if File.exist?(icuconfig) && `#{icuconfig} --cxxflags`.include?("c++11")
+  $CXXFLAGS << ' -std=c++11'
```

Comment:
Reading the source of mkmf.rb, it appears these are always initialized as strings, and should never be nil: https://github.com/ruby/ruby/blob/trunk/lib/mkmf.rb#L2507-L2529